moveos/moveos-types/src/list_access_path.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::object::{NamedTableID, ObjectID};
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum PathList {
+    /// Get account resources list
+    Resource { account: AccountAddress },
+    /// Get account modules list
+    Module { account: AccountAddress },
+    /// Get table values list by table handle
+    Table { table_handle: ObjectID },
+}
+
+impl std::fmt::Display for PathList {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PathList::Resource { account } => {
+                write!(f, "/resource/{}", account.to_hex_literal(),)?;
+            }
+            PathList::Module { account } => {
+                write!(f, "/module/{}", account.to_hex_literal(),)?;
+            }
+            PathList::Table { table_handle } => {
+                write!(f, "/table/{}", table_handle,)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for PathList {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.trim_start_matches('/');
+        let mut iter = s.split('/');
+        let path_type = iter.next().ok_or_else(|| anyhow::anyhow!("Invalid path"))?;
+        match path_type {
+            "resource" => {
+                let account = iter.next().ok_or_else(|| anyhow::anyhow!("Invalid path"))?;
+                let account = AccountAddress::from_hex_literal(account)?;
+                Ok(PathList::Resource { account })
+            }
+            "module" => {
+                let account = iter.next().ok_or_else(|| anyhow::anyhow!("Invalid path"))?;
+                let account = AccountAddress::from_hex_literal(account)?;
+                Ok(PathList::Module { account })
+            }
+            "table" => {
+                let table_handle = iter.next().ok_or_else(|| anyhow::anyhow!("Invalid path"))?;
+                let table_handle = ObjectID::from_str(table_handle)?;
+                Ok(PathList::Table { table_handle })
+            }
+            _ => Err(anyhow::anyhow!("Invalid path: {}", s)),
+        }
+    }
+}
+
+/// StateDB list query path
+///
+/// 1. /resource/$account_address
+/// 2. /module/$account_address
+/// 3. /table/$table_handle
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AccessPathList(pub PathList);
```

Comment:
How about directly use the AccessPath? AccessPathList = AccessPath without keys.